src/Api/Vault/Validators/CipherRotationValidator.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Bit.Api.Auth.Validators;
+using Bit.Api.Vault.Models.Request;
+using Bit.Core.Entities;
+using Bit.Core.Exceptions;
+using Bit.Core.Vault.Entities;
+using Bit.Core.Vault.Repositories;
+
+namespace Bit.Api.Vault.Validators;
+
+public class CipherRotationValidator : IRotationValidator<IEnumerable<CipherWithIdRequestModel>, IEnumerable<Cipher>>
+{
+    private readonly ICipherRepository _cipherRepository;
+
+    public CipherRotationValidator(ICipherRepository cipherRepository)
+    {
+        _cipherRepository = cipherRepository;
+    }
+
+    public async Task<IEnumerable<Cipher>> ValidateAsync(User user, IEnumerable<CipherWithIdRequestModel> ciphers)
```

Comment:
⛏️ I'm not a huge fan of having multiple return statements. 

```suggestion
public async Task<IEnumerable<Cipher>> ValidateAsync(User user, IEnumerable<CipherWithIdRequestModel> ciphers)
{
    var result = new List<Cipher>();
    if (ciphers != null || ciphers.Any())
    {
        var existingCiphers = await _cipherRepository.GetManyByUserIdAsync(user.Id);
        if (existingCiphers != null || existingCiphers.Any())
        {
            foreach (var existing in existingCiphers)
            {
                var cipher = ciphers.FirstOrDefault(c => c.Id == existing.Id);
                if (cipher == null)
                {
                    throw new BadRequestException("All existing ciphers must be included in the rotation.");
                }
                result.Add(cipher.ToCipher(existing));
            }
        }
    }

    return result;
}
```

Reply:
I'm actually a pretty big fan of early returns, which is intended to avoid more than a couple levels of nesting. I think it makes the logic harder to follow because you have to keep in mind all the previous operations.